includes/class-search-categories-metabox.php

```diff
@@ -0,0 +1,281 @@
+		$this->plugin_file = $file;
+	}
+
+	/**
+	 * Gets the Post taxonomy Category slug by Post ID.
+	 *
+	 * @since 0.1.0
+	 * @access private
+	 *
+	 * @param $post_id
+	 * @return array
+	 */
+	private function get_post_taxonomy( $post_id ) {
+		$taxonomies = get_post_taxonomies( $post_id );
+		$taxonomy   = $taxonomies[0];
+		return $taxonomy;
+	}
+
+	/**
+	 * Saves the Post Categories to DB.
+	 *
+	 * @since 0.1.0
+	 */
+	public function save_post_categories() {
+		global $post;
+
+		$post_taxonomy       = $this->get_post_taxonomy( $post->ID );
+		$selected_categories = $_POST['post-category'];
+		if ( isset( $selected_categories ) && ! empty( $selected_categories ) ) {
+			$selected_categories = array_map( 'intval', $selected_categories );
+			$selected_categories = array_unique( $selected_categories );
+			wp_set_object_terms( $post->ID, $selected_categories, $post_taxonomy, false );
+		}
+	}
+
+	/**
+	 * Gets the array of categories.
+	 *
+	 * @since 0.1.0
+	 * @access private
+	 *
+	 * @param string $search_term
+	 * @return array
+	 */
+	protected function get_taxonomy_categories( $search_term = '' ) {
+		$categories = array();
+		$args       = array();
+		$taxonomy   = array();
+
+		/*
+		 * Get only Category taxonomy
+		 * Refer https://codex.wordpress.org/Function_Reference/get_post_taxonomies
+		 */
+		$taxonomy   = $this->get_post_taxonomy();
+
+		$args = array(
+			'hide_empty' => false,
+			'order_by'   => 'term_id',
+			'order'      => 'ASC',
+			'taxonomy'   => $taxonomy,
+		);
+
+		if ( trim( $search_term ) != '' ) {
+
+			$additional_args = array(
+				'name__like' => $search_term,
+				'number'     => 10,
+			);
+
+			$args = wp_parse_args( $additional_args, $args );
+		}
+
+		$categories = get_categories( $args );
+		return $categories;
+	}
+
+	/**
+	 * Renders the Category div in the posts editor.
+	 *
+	 * @since 0.1.0
+	 */
+	public function search_category_div_callback() {
+	?>
+
+		<p>
+			<input id="scm-search-categories-search" type="text" name="scm-search-categories-search" size="16" />
+			<input type="button" class="button" id="scm-search-clear" value="Clear" />
+		</p>
+
+		<div class="scm-categories-panel">
+			<ul id="scm-categories-ul">
+			<?php
+			global $post;
+			$categories = $this->get_taxonomy_categories();
+			$post_taxonomy = $this->get_post_taxonomy( $post->ID );
+			$post_categories = get_the_terms( $post->ID, $post_taxonomy );
+			$post_categories = array_column( $post_categories, 'term_id' );
+			foreach( $categories as $category ) :
+				$is_post_in_category = in_array( $category->term_id, $post_categories, true );
+			?>
+				<li id='<?php echo $category->term_id; ?>'>
+					<label class="selecttit">
+						<input id='in-category-<?php echo $category->term_id; ?>' class="scm-category-list-item" value=<?php echo $category->term_id; ?> name="post-category[]" type="checkbox" <?php if ( $is_post_in_category ) { echo 'checked="checked"'; } ?>> <?php echo esc_html( $category->name ); ?>
```

Comment:
Instead of this condition `if ( $is_post_in_category ) { echo 'checked="checked"'; }` you can use the `checked` function